omc3/scripts/bad_bpms_summary.py

```diff
@@ -0,0 +1,412 @@
+    if outfile is not None:
+        tfs.write(outfile, df_evaluated)
+
+    if opt.print_percentage is not None:
+        print_results(df_evaluated, opt.print_percentage)
+
+
+# Collection of Data ---
+
+def get_empty_df() -> tfs.TfsDataFrame:
+    """ Create an empty TfsDataFrame with the correct column names. """
+    return tfs.TfsDataFrame(columns=[NAME, ACCEL, PLANE, SOURCE, FILE])
+
+
+def collect_bad_bpms(root: Path, dates: Sequence[Path | str], accel_glob: str) -> tfs.TfsDataFrame:
+    """ Create a TfsDataFrame with all bad-bpms within selected dates.
+
+    Args:
+        root (Path): Root path to the GUI output folder.
+        dates (Sequence[Path | str]): List of dates or glob patterns to collect bad-bpms from.
+        accel_glob (str): Accelerator name (glob for the sub-directories).
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms within selected dates.
+    
+    """
+    dfs = []
+
+    def collect_and_append(date_dir: Path):
+        """ Helper to collect for date_dir and append to dfs if not None. """
+        df_new = collect_date(date_dir, accel_glob)
+        if df_new is not None:
+            dfs.append(df_new)
+
+    # Loop over dates ---
+    for date in dates:
+        date_dir = root / date
+        if date_dir.is_dir():
+            collect_and_append(date_dir)
+
+        else:  
+            for date_dir in root.glob(date):
+                collect_and_append(date_dir)
+    
+    # Check and return ---
+    if not len(dfs):
+        LOG.warning("No bad-bpms found! Resulting TfsDataFrame will be empty.")
+        return get_empty_df()
+
+    return tfs.concat(dfs, axis="index", ignore_index=True) 
+
+
+def collect_date(date_dir: Path, accel_glob: str) -> tfs.TfsDataFrame | None:
+    """ Collect bad-bpms for a single date, by checking the sub-directories 
+    which conform to the `accel_glob` pattern.
+
+    In each accel directory, check for sub-directories named `Measurements` and `Results`,
+    which in turn contain which in turn have entries containing the bad-bpms files.
+    
+    Args:
+        date_dir (Path): Path to the date directory.
+        accel_glob (str): Accelerator name (glob for the sub-directories).
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms for the date.
+    """
+    dfs: list[tfs.TfsDataFrame] = []
+
+    for accel_dir in date_dir.glob(accel_glob):
+        for subdir_name in (MEASUREMENTS_DIR, RESULTS_DIR):
+            analysis_stage_dir = accel_dir / subdir_name 
+            if not analysis_stage_dir.is_dir():
+                continue
+
+            for data_dir in analysis_stage_dir.iterdir():
+                if not data_dir.is_dir():
+                    continue
+
+                df_collected = collect_bad_bpm_files_in_dir(data_dir)
+                if df_collected is not None:
+                    df_collected.loc[:, ACCEL] = accel_dir.name
+                    dfs.append(df_collected)
+
+    if not len(dfs):
```

Comment:
Correct me if I'm wrong, but would `not dfs` be better instead here?

Reply:
but you will find different flavours of this depending on who writes the code